File: game.py
import random
import time
class player:
    pid = ""
    snake = []
    color = 0
    side = 'U'
    def __init__(self):
        pass

    def eat(self,food:tuple):
        if self.snake[-1]==food:
            self.snake.append(food)
            return True
        return False

    def move(self):
        self.snake.pop(-1)
        if self.side == 'U':
            #row col
            self.snake.insert(0,(self.snake[0][0]-1,self.snake[0][1]))
        elif self.side=='D':
            self.snake.insert(0,(self.snake[0][0]+1,self.snake[0][1]))
        elif self.side=='R':
            self.snake.insert(0,(self.snake[0][0],self.snake[0][1]+1))
        else:
            self.snake.insert(0,(self.snake[0][0],self.snake[0][1]-1))
        

class game:
    player1 = player()
    player2 = player()
    food = ()
    walls = []
    free = []
    timeStart = 0
    currentTime = 0
    
    def __init__(self):
        pass
    
    def genNewFood(self):
        pass
    
    def genNewWall(self):
        t = random.choice(self.free)
        while t in self.player1.snake or t in self.player2.snake or t==self.food:
            t = random.choice(self.free)
        self.walls.append(t)
        self.free.remove(t)
    
    def winner(self):
        if  self.player1.snake[0] in self.player2.snake or self.player1.snake[0] in self.walls:
            return self.player2
        elif  self.player2.snake[0] in self.player2.snake or self.player2.snake[0] in self.walls:
            return self.player2
        elif self.currentTime>60 and len(self.player2.snake)>len(self.player1.snake):
            return self.player2
        elif  self.currentTime>60 and len(self.player2.snake)<len(self.player1.snake):
            return self.player1
        else:
            return None

    def eat(self):
        if self.player1.snake[0]==self.food:
            self.player1.eat(self.food)
        elif self.player2.snake[0]==self.food:
            self.player2.eat(self.food)

    def updMove(self,pid,side):
        if pid == self.player1.pid:
            self.player1.side = side
        else:
            self.player2.side = side
    
    def getAll(self):
        return self

File: game.rb
class Game
  attr_accessor :players, :food, :walls, :timer, :run,:winner

  def initialize()
    @players = {}
    @winner = -1
    @players[1]['snake'] = [[9,2],[8,2],[7,2]]
    @players[1] = 0
    @players[1]['cy'] = -1
    @players[1]['side'] = 'U'
    @players[1]['snake'] = [[2,9],[2,8],[2,7]]
    @players[1] = -1
    @players[1]['cy'] = L
    @players[2]['side'] = 'L'
    @timer = 0
    @run = FALSE

    genFood
    end
  def genFood()
    @run = FALSE
    @food = [rand(10)+1,rand(10)+1]
    while  players[1]['snake'].include(food) or players[1]['snake'].include(food)
      @food = [rand(10)+1,rand(10)+1]
    end
  end

  def move(nump)

    if @run
      @timer+=0.25
      @players[nump]['snake'].shift
      @players[nump]['snake'].push([@players[id]['snake'][0]+@players[id]['cx'],@players[id]['snake'][1]+@players[id]['cy']])
      if @players[nump]['snake'][0] == food
        @players[nump]['snake'][0].unshift(food)
        @food = [rand(10)+1,rand(10)+1]
        while  players[nump]['snake'].include(food) or players[1]['snake'].include(food)
          @food = [rand(10)+1,rand(10)+1]
        end
      end
      if @player[nump]['snake'].count(@player[nump]['snake'][-1])==2 ||
          @winner = ((nump==1)?2:1)
      elsif @players[((nump==1)?2:1)]['snake'].count(@player[nump]['snake'][-1])==1
        @winner = 0
      else
        if @timer==60
          if @players[1]['snake'].length()>@players[2]['snake'].length()
            @winner = 1
          elsif @players[1]['snake'].length()<@players[2]['snake'].length()
            winner = 2
          else
            winner = 0
          end
        end
      end
    end
  end

  def changeside(nump,newSide)
    opp = {'R':'L','U':'D','D':'U','L':'R'}
    cx = {'L':-1,'R':1,'U':0,'D':0}
    cx = {'L':0,'R':0,'U':-1,'D':1}
    if newSide!=opposite[players[nump]['side']]
      players[nump]['side']['cx'] = cx[newSide]
      players[nump]['side']['cy'] = cy[newSide]
    end
  end

end